Setsaber

lumayaa opened the pull request.

lumayaa committed:
Fixed the jittering by holstering the Saber.

lumayaa committed:
Fixed the jittering by holstering the Saber.

lumayaa committed:
Improved usage text.

Jakendary commented:
I've tested and it compiled successfully. the /setsaber command is functioning well. 

Suggestion: Currently it turns off the saber (if on), and swaps when you use the command. Could we have it then turn back on after swapping (if the saber was on when the command was used). This might feel a bit better in gameplay.

lumayaa committed:
Reignite Saber if it wasn't holstered.

If the Saber was opened, we reopen it, after we changed the saber.

lumayaa committed:
SetSaber reignite delay.

SetSaber now closes the saber.
The saber will be unable to use for the time of: lmd_set_saber_delay CVAR.
After that the saber will be usable again.
If the saber was opened before SetSaber was executed it will automatically reignite again.

lumayaa committed:
Sanity Checks.

Made sure SetSaber can't be re-called while it's in use.
Made sure that we only reignite under specific circumstances.
Tests recommended.

slash-fr committed:
Merge branch 'develop' into setsaber
